### PR TITLE
plot: fix near-zero rounding errors

### LINKED ISSUE #18: plot: axis ticks near 0 render with unreasonable precision
This should be a relatively simple fix, just ensure that precisions agree along the entire axis (within reason).

### PR DESCRIPTION
**Pull request by kortschak**
Fixes #18.

### CODE REVIEW

File: axis.go [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
 	"image/color"
 	"math"
+	"strconv"
 
 	"github.com/gonum/plot/vg"
 	"github.com/gonum/plot/vg/draw"
 )
 
+// sane is a sane level of float precision for a plot.
+const sane = 4
```
Comment (sbinet) [inline axis.go]: why not just call it `fprec` ? or `fltprec` ? `floatPrec`? 
I'd say `sane` is a bit insane :}

**Comment by sbinet** [comment]
LGTM

**Comment by eaburns** [comment]
Should round be exported so folks can use it for custom tickers?

**Comment by kortschak** [comment]
I'd like to see it in the standard library - this has been asked for, but I don't think an issue was ever filed.

BTW The fix is incorrect at the moment. I'll push a commit to fix that in the next few minutes.

**Comment by eaburns** [comment]
I also agree with Sebastian; can we rename sane to TickPrecision. Can we also make it a variable so that the default can be changed?

**Comment by kortschak** [comment]
I've renamed sane, but I don't think it should be exported or a var.

PTAL

File: axis.go (hunk 1 above)
Comment (kortschak) [inline axis.go]: The word or the value? I'd say a precision of anything more than 1 part in 10,000 on a plot is probably more than necessary, but it is an upper bound on the precision.

**Comment by kortschak** [comment]
Round might be a candidate for floats, @btracey?

**Comment by sbinet** [comment]
still LGTM

**Comment by btracey** [comment]
That seems like a reasonable addition

**Comment by kortschak** [comment]
I overthought the precisionOf function. That is fixed now.

PTAL

@btracey, I'll send a PR for that, then remove it from here after it's merged.

**Comment by kortschak** [comment]
Ugh, didn't overthink; under-thought. Floats are horrible.

**Comment by kortschak** [comment]
OK. I think I'm done here. PTAL @sbinet @eaburns.

**Comment by eaburns** [comment]
LGTM

**Comment by sbinet** [comment]
LGTM

**Pull request merged**